### ISSUE TITLE
How to read a remote pairlist

### ISSUE THREAD

**Issue by WinPoss** [freqtrade/freqtrade#10358]
## Describe your environment

* Operating system: ubuntu 20.04.6
* Python Version: 3.12
* CCXT version: 4.3.11
* Freqtrade Version: 2024.4

  
## Your question

*Ask the question you have not been able to find an answer in the [Documentation](https://www.freqtrade.io/en/latest/)*

I would like to achieve the following effect: I want to retrieve the asset list from a remote server's asset API endpoint. Each time I fetch new data, I want it to overwrite the previous asset list.

For example, if the first retrieval fetches 15 assets, and the second retrieval fetches 10 assets, I want the second retrieval to directly replace the previous 15 assets.

I have looked at the official documentation regarding the [RemotePairList](https://www.freqtrade.io/en/stable/plugins/#remotepairlist), and it seems that there are only two modes available: `filter` and `append`. If I understand correctly, the `filter` mode retrieves the intersection of the first and second lists, while the `append` mode retrieves the union of both lists.

Is there any way to achieve the behavior of directly overwriting the data from the first retrieval when fetching the second time?

Thank you.

**Comment by stash86** [comment]
Second pairlist will always act as filter. In your case, if first pairlist fetch 15, then if you want only 10 at second pairlist, you need to find a way to filter the bad 5 to be removed.

If you want second filter to fetch 10 then replace the first pairlist, what is the purpose of having the first pairlist then? You can just remove the first pairlist, and move the second pairlist into first position.

**Comment by WinPoss** [comment]
**You can just remove the first pairlist, and move the second pairlist into first position.**

How is this achieved? Is it possible to remove the first read list in the strategy code?

**If you want second filter to fetch 10 then replace the first pairlist, what is the purpose of having the first pairlist then?** 

Because the asset's growth ranking in the market will change, I hope to get a new market ranking every 30 minutes. At this time, the list ranking read for the first time does not meet the requirements, and the second new ranking needs to be used.

**Comment by stash86** [comment]
Well, you need to learn programming and stop relying on chatgpt. It's doable to do what you want to do, but you will need to learn to do it yourself. You need to learn english, and read the ccxt/exchange docs whether you can get such data.

If you just want first pairlist to refresh data, not sure why you need second pairlist? You can always ask the first pairlist to refresh new data. Please take time to understand how freqtrade work, how pairlist work. Read the source code. Try to understand them. Throwing them to chatgpt won't help you understand it better.

**Comment by WinPoss** [comment]
Okay, thank you for your suggestion. Fortunately, I am very interested in quantitative trading, otherwise it would be difficult to persevere.

Regarding the content you mentioned about refreshing data, are you saying that if I set the refresh_period parameter in the RemotePairList, for example:

```
"pairlists": [
    {
        "method": "RemotePairList",
        "mode": "whitelist",
        "processing_mode": "filter",
        "pairlist_url": "https://example.com/pairlist",
        "number_assets": 10,
        "refresh_period": 1800,
        "keep_pairlist_on_failure": true,
        "read_timeout": 60,
        "bearer_token": "my-bearer-token",
        "save_to_file": "user_data/filename.json" 
    }
]
```

Then, in fact, the list data saved in the bot will be automatically refreshed every 30 minutes, and the newly obtained remote data will be used directly without being affected by the filter and append, right?

**Comment by xmatthias** [comment]
Yeah correct.

That "second pairlist" topic was a bit confusing, as in reality, you dont mean the 2nd pairlist, but instead a new call to the i itial pairlist.

A remotepairlist as generator (the first in the pairlist config) will do just that, replace what was there before.

**Comment by WinPoss** [comment]
I have achieved the effect I wanted. I did one thing even though I didn’t understand many techniques, but I still did it.

There are several tips for setting up RemotePairList. Record them in the hope that users who need this in the future can see them.

1. `"pairlist_url": "https://example.com/pairlist",` can be a local JSON file.

2. The path mentioned above does not start with `/root`, but is in the following format: `"pairlist_url": "file:////freqtrade/user_data/get_pairlist/pairlist.json"`.

3. I did not use a remote server. I wrote a Python program in the server running the robot and created a server scheduled task. After getting the list of trading pairs, I wrote it directly into the above file, but this step requires an absolute path starting with `/root`.

I was finally able to use freqtrade to implement some of my strategy logic, performing virtual disks for a while first, but I'm still very happy.

Thank you, freqtrade.
